mountpoint-s3/src/cli.rs

```diff
@@ -0,0 +1,250 @@
+        conflicts_with="read_only",
+    )]
+    pub read_write: bool,
+    #[clap(long = "", default_value = "true", hide = true)]
+    pub foreground: bool,
+}
+
+pub fn get_cli_args() -> anyhow::Result<CliArgs> {
+    let is_fstab = env::args_os().len() == 5 && env::args_os().nth(3) == Some("-o".parse().unwrap());
+
+    if is_fstab {
+        let args = FsTabCliArgs::parse();
+        args.try_to_cli_args()
+    } else {
+        let args = AppCliArgs::parse();
+        Ok(args.cli_args)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use proptest::{prop_assert_eq, proptest};
+    use test_case::test_case;
+
+    #[test_case("no commas", Some(["no commas"].to_vec()))]
+    #[test_case("simple,case", Some(["simple", "case"].to_vec()))]
+    #[test_case("comma\\,escape", Some(["comma,escape"].to_vec()))]
+    #[test_case("a,\\,,c", Some(["a", ",", "c"].to_vec()))]
+    #[test_case("a,\\,\\\",d", Some(["a", ",\"", "d"].to_vec()))]
+    #[test_case("£,c", Some(["£", "c"].to_vec()))]
+    #[test_case("\\\"", Some(["\""].to_vec()))]
+    #[test_case("--read-only", Some(["--read-only"].to_vec()))]
+    #[test_case("ends,with,\\", None)]
+    #[test_case("random,\"quotes\"", None)]
+    #[test_case("\\e\\scaped,\\w\\rong", None)]
+    fn test_split_commas(string: &str, expected: Option<Vec<&str>>) {
+        assert_eq!(
+            FsTabCliArgs::split_commas(string).ok(),
+            expected.map(|v| v.iter().map(|&x| x.into()).collect())
+        );
+    }
+
+    #[test_case(["_", "demo_s3_bucket", "/mnt/test", "-o", ""].to_vec())]
+    #[test_case(["_", "demo_s3_bucket", "/mnt/test", "-o", "_netdev"].to_vec())]
+    #[test_case(["_", "demo_s3_bucket", "/mnt/test", "-o", "ro"].to_vec())]
+    #[test_case(["_", "demo_s3_bucket", "/mnt/test", "-o", "rw"].to_vec())]
+    #[test_case(["_", "demo_s3_bucket", "/mnt/test", "-o", "uid=2,gid=4,debug"].to_vec())]
+    #[test_case(["_", "demo_s3_bucket", "/mnt/test", "-o", "prefix=foo/bar\\,baz/"].to_vec())]
+    fn test_fstab_cli_args_parses(args: Vec<&str>) {
+        let fstab_cli_args = FsTabCliArgs::try_parse_from(&args).unwrap();
+        let _ = fstab_cli_args.try_to_cli_args().unwrap();
+    }
+
+    #[test]
+    fn test_fstab_cli_args_parses_correctly() {
+        let args = [
+            "_",
+            "demo_s3_bucket",
+            "/mnt/test",
+            "-o",
+            "prefix=foo/bar\\,baz/,_netdev,uid=2,ro",
+        ]
+        .to_vec();
+        let fstab_cli_args = FsTabCliArgs::try_parse_from(args).unwrap();
+        let cli_args = fstab_cli_args.try_to_cli_args().unwrap();
+
+        assert!(cli_args.foreground);
+        assert_eq!(cli_args.base.mount_point.to_str(), Some("/mnt/test"));
+        assert_eq!(cli_args.base.bucket_name, "demo_s3_bucket");
+        assert_eq!(cli_args.base.prefix.unwrap().as_str(), "foo/bar,baz/");
+        assert_eq!(cli_args.base.uid, Some(2));
+        assert_eq!(cli_args.base.gid, None);
+        assert!(cli_args.read_only);
```

Comment:
What happens if I do `ro=recursive` or `ro=vfs`?

I think on this point we want to document the intent here: we take options, parse everything we recognise and support(!) and then reject anything we can't support. I'd even add a test case to cover what happens if a customer provides `ro=vfs`. What do negative cases look like for customers trying to configure things they shouldn't?